api/oc_discovery.c

```diff
@@ -379,9 +403,21 @@ filter_oic_1_1_resource(oc_resource_t *resource, oc_request_t *request,
 
   // port, x.org.iotivity.tcp and x.org.iotivity.tls
   oc_endpoint_t *eps = oc_connectivity_get_endpoints(resource->device);
+  size_t device_index = resource->device;
+#ifdef OC_SECURITY
+  bool owned_for_SVRs =
+    (oc_core_is_SVR(resource, device_index) &&
+     (((oc_sec_get_pstat(device_index))->s != OC_DOS_RFOTM) ||
+      oc_tls_num_peers(device_index) != 0));
+#else  /* OC_SECURITY */
+  bool owned_for_SVRs = false;
+#endif /* OC_SECURITY */
+
   while (eps != NULL) {
```

Comment:
What about switching while cycle to for, so the assignment of next is writen just once:
```
for(; eps!=NULL; eps = eps->next)
```